Pass on communication section (#769)
browserengineering/book: book/embeds.md

```

We'll add each window to the global map as it's created:

``` {.python}
class JSContext:
    def add_window(self, frame):
        # ...
        self.interp.evaljs("WINDOWS[{}] = window_{};".format(
            frame.window_id, frame.window_id))
```

Now when you call `window.parent`, we can look up the correct `Window`
object in this global map:

``` {.html}
Object.defineProperty(Window.prototype, 'parent', {
  configurable: true,
  get: function() {
    var parent_id = call_python('parent', window._id);
    if (parent_id != undefined) {
        var parent = WINDOWS[parent_id];
        if (parent === undefined) parent = new Window(-1);
        return parent;
    }
  }
});
```

Note that it's possible for the lookup in `WINDOWS` to fail, if the
parent frame is not in the same origin as the current one and
therefore isn't running in the same `JSContext`. In that case, this
code return a fresh `Window` object with a fake, negative id. This
makes further accesses to that window's APIs to fail, making it
impossible to modify that frame's state.^[Note however that in a real
browser, this `Window` object is not fake, and some APIs can be called

> on it, most important its `parent` can also be retrieved. There is a

Reviewer: ```suggestion
on it, such as that its `parent` can also be retrieved. There is a
```

}
```

> That's everything on the receiver side; now let's do the sender side.

Reviewer: ```suggestion
That's everything on the receiver side; now let's do the sender side and implement
```
